packages/errors/src/index.ts

```diff
@@ -41,15 +41,11 @@ export interface ErrorWithOptionalCode extends Error {
   code?: string;
 }
 
-export interface ErrorWithName extends Error {
-  name: string;
-}
```

Comment:
So the reasoning for removing this and not checking name at all is that the aws docs led us to believe that `code` is always reliable and `name` is descriptive and can change? Are there docs to support that/do I have it right?

Reply:
ah, yeah my mistake. Words are hard.

What about _only_ checking `name`? Is there value in continuing to evaluate `code` if the sdk doesn't recommend using it (that's right, right?)? Checking both is probably fine even if it's not necessary but want to be sure we're sure which field they want us to use.

Reply:
Ok, it would be nice to be more confident but it's also not a big deal to check both. The event that has code mean one thing and name mean the exact opposite seems unlikely 😆